### PR TITLE
Enable to embed external vector graphics

### PR DESCRIPTION
**Pull request by korintje**
This PR adds the function to import a .svg file as a vector image, not a raster image.

Veusz has powerful figure composing ability, in addition to the excellent plotting functions. Therefore, I think some users will have benefit if veusz can import an external vector-based images (e.g. chemical structures) without rasterizing. Although the "add imagefile" function also can import .svg vector graphics, it rasterizes the images and sometimes makes their resolutions low.

This PR implements `SVGFile` class, which is similar to `ImageFile` class.
- The `SVGFile` instances can be added by "Add svgfile" button in the shape tools menu.
- The `SVGFile` instances are rendered by `QSvgRenderer`.
- The `SVGFile` instances are encoded into base64 data with type of `svgfile`.

The imported image can be rendered and embedded as a vector image as shown in the picture below.

`SVGFile` is implemented separately from `ImageFile` class to keep the backward compatibility and reduce unnecessary conditional branches. The newly added code in the `veusz.widgets.shape` may need to have more sophisticated (many of the method in the `SVGFile` and `ImageFile` are duplicated). I left it as is so far because I am new to the veusz policy for code design (e.g. Are function overloads discouraged or not?). I think there are some options. I would like to discuss how to improve it. 

I would appreciate if you consider adding this function.

Many Thanks!
korintje

### CODE REVIEW

**Comment by korintje** [comment]
Although I have informed that build checks at Ubuntu were unsuccessful, it seems to be not related to my file changes.
It may be derived from some dead links of libraries for Ubuntu 18.0x `apt-get` commands.

**Comment by jeremysanders** [comment]
Thanks for the contribution. I'll take a look. The build is failing because I didn't have an `apt-get update` in the build commands. That should be fixed if you merge with the latest master branch.

**Comment by korintje** [comment]
I have merged with the latest master branch, and checks have passed.
Thanks for the suggestion.

File: veusz/setting/controls.py [hunk 1]
```diff
@@ -1790,6 +1790,10 @@ def buttonClicked(self):
             title = _('Choose image')
             filefilter = ("Images (*.png *.jpg *.jpeg *.bmp *.svg *.tiff *.tif "
                           "*.gif *.xbm *.xpm);;" + filefilter)
+        
+        elif self.mode == 'svg':
+            title = _('Chose SVG file')
```
Comment (jeremysanders) [inline veusz/setting/controls.py:1795]:
```suggestion
            title = _('Choose SVG file')
```

File: veusz/widgets/shape.py [hunk 2]
```diff
@@ -382,6 +382,155 @@ def drawShape(self, painter, rect):
...
+
+        if s.filename == '{embedded}':
+            print("Data already embedded")
+            return
+
+        # get data from external file
+        try:
+            f = open(s.filename, 'rb')
+            data = f.read()
+            f.close()
```
Comment (jeremysanders) [inline veusz/widgets/shape.py:442]:
```suggestion
            with open(s.filename, 'rb') as f:
                data = f.read()
```

**Review by jeremysanders — CHANGES_REQUESTED** [review]
This looks a very nice addition to the program. Thanks a lot!

I have a couple of very minor suggestions (hopefully you can see these) - there's a typo and I suggest using a with statement for reading the file.

I can fix these after merging, or feel free to make the suggested changes I've made.

Thanks again

Jeremy

**Review by jeremysanders — APPROVED** [review]

**Comment by korintje** [comment]
Thanks for the review!
I have merged changes you suggested.

**Comment by jeremysanders** [comment]
Thanks very much for the contribution - merged!

**Pull request merged**